src/dex/balancer-v2/LinearMath.ts

```diff
@@ -0,0 +1,299 @@
+import { MathSol } from './balancer-v2-math';
+
+type Params = {
+  fee: bigint;
+  lowerTarget: bigint;
+  upperTarget: bigint;
+};
+
+export function _calcBptOutPerMainIn(
+  mainIn: bigint,
+  mainBalance: bigint,
+  wrappedBalance: bigint,
+  bptSupply: bigint,
+  params: Params,
+): bigint {
+  // Amount out, so we round down overall.
+
+  if (bptSupply == BigInt(0)) {
+    return _toNominal(mainIn, params);
+  }
+
+  const previousNominalMain = _toNominal(mainBalance, params);
+  const afterNominalMain = _toNominal(mainBalance + mainIn, params);
+  const deltaNominalMain = afterNominalMain - previousNominalMain;
+  const invariant = _calcInvariantUp(previousNominalMain, wrappedBalance);
+  return MathSol.divDownFixed(
+    MathSol.mulDownFixed(bptSupply, deltaNominalMain),
+    invariant,
+  );
```

Comment:
I am checking against the implementation https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/pool-linear/contracts/LinearMath.sol#L75
Shouldn't this be? 
```suggestion
  return MathSol.divDown(
    MathSol.mul(bptSupply, deltaNominalMain),
    invariant,
  );
```